### ISSUE TITLE
Unable to find registered backend

### ISSUE THREAD

**Issue by anLizard** [itzg/mc-router#9]
I get instantly disconnected upon trying to connect to a routed server.

docker-compose log after trying to connect to first.example.com:
`mc-router  | time="2020-05-08T17:58:41Z" level=warning msg="Unable to find registered backend" serverAddress=example.com.`

Minecraft client error is just "Disconnected".
I opened port 25565 and made sure the A records on my DNS exist.
I can connect to both servers if I map them to two different open ports without the router.

docker-compose.yml:
```
version: '3'

services:

  router:
    image: itzg/mc-router
    container_name: mc-router
    ports:
      - 25565:25565
    command: --mapping=first.example.com=first:25565,second.example.com=second:25565

  first:
    image: itzg/minecraft-server
    container_name: mc-first
    expose:
      - "25565"
    volumes:
      - ./first:/data
      - /etc/localtime:/etc/localtime:ro
    environment:
      EULA: "TRUE"
    restart: always

  second:
    image: itzg/minecraft-server
    container_name: mc-second
    expose:
      - "25565"
    volumes:
      - ./second:/data
      - /etc/localtime:/etc/localtime:ro
    environment:
      EULA: "TRUE"
    restart: always
```

**Comment by anLizard** [comment]
Nevermind. Turns out I had an SRV record that routed to just example.com isntead of first.example.com
Removing that fixed the problem for me

**Comment by zmweske** [comment]
Could you post some more information about how you have your dns records set up? My goal is to use SRV records to direct _minecraft._tcp.first.example.com to a non-standard port. I can only get it to work when I explicitly type the port in the game (meaning first.example.com does not work but first.example.com:port does). The SRV record is supposed to eliminate the need for this. Additionally, I am only able to do this with one level of subdomains. I cannot do first.mc.example.com or second.mc.example.com regardless of any DNS settings or specifying the nonstandard ports. @itzg I am curious if you have any thoughts on this as well. Am I asking too much of this application or is this unexpected behavior?
Thanks,
-Zach

**Comment by itzg** [comment]
I would recommend not using SRV records when using mc-router, bungee, etc since the host to port multiplexing is being done by these. In other words, using SRVs is totally fine, but you don't need to use mc-router then.

**Comment by zmweske** [comment]
So I basically shouldn't use a non-standard port? I know security through obscurity is not necessarily a perfect practice but I would rather have the servers behind a reverse proxy on a non-standard port so that bots are not able to scan as effectively. Granted, you would still need the correct subdomain for it to work anyways.

**Comment by itzg** [comment]
Interesting, I didn't think about that being a use of SRV record there. Will need to think about the issue some more then.

**Comment by itzg** [comment]
...and btw the Minecraft client itself should be resolving the SRV, so maybe use Witeshark on your client machine to see what it's doing there DNS wise and then port wise.

**Comment by zmweske** [comment]
if others are interested, I was able to get it working with https://github.com/haveachin/infrared
- a DNS SRV record _minecraft._tcp.mc.example.com to port 12345 (with CNAME for mc.example.com)
- the firewall rule/NAT routing 12345 to 12345
- docker mapping the host's 12345 port to 25565 inside the container

**Comment by zmweske** [comment]
I attempted another crack at getting things working. This was all using the mappings mc.example.com=192.168.1.x:25566 and 192.168.1.x=192.168.1.x:25566. The mc-router binds 12345:25565 and 25565:25565. 

Connecting in game, I was able to log on using 192.168.1.x and mc.example.com:12345. What still didn't work, was mc.example.com without the specified port. It is able to query the server for all 3 access methods, though attempting to connect using just the domain gives the disconnected message.
